### PR TITLE
Switch from deprecated ReactiveCoca version to ReactiveObjC

### PR DESCRIPTION
**Pull request by DinosaurDad**
ReactiveCocoa has changed to Swift, so moving ARAnalytics to ReactiveObjC is a better fit. This allows apps that have moved to ReactiveObjC to avoid also including ReactiveCocoa when using the DSL subspec.

### CODE REVIEW

**Comment by orta** [comment]
Any chance you can run `bundle update color2`?

**Comment by orta** [comment]
That should fix the red

**Comment by DinosaurDad** [comment]
Not sure how to proceed. Took a few stabs at fixing the `danger` gem issues, but I am unfamiliar with it and don't know how to run it locally. The last build doesn't show any detail around why it errors out.

```
$ bundle exec danger
Danger has failed this build. 
Found 1 error.
```

Suggestions?

**Comment by orta** [comment]
Oh damn, that's a lot of commits - @Matt-ExpeData you don't need to make this go green, that's my responsibility - I was kinda hoping it would just require a quick fix.

File: ARAnalytics.podspec [hunk 1]
```diff
@@ -5,7 +5,7 @@ Pod::Spec.new do |s|
   s.homepage     =  'https://github.com/orta/ARAnalytics'
   s.authors      =  { 'orta' => 'orta.therox@gmail.com', 'Daniel Haight' => "confidence.designed@gmail.com" }
   s.source       =  { :git => 'https://github.com/orta/ARAnalytics.git', :tag => s.version.to_s }
-  s.ios.deployment_target = "7.0"
+  s.ios.deployment_target = "8.0"
```
Comment (orta) [inline ARAnalytics.podspec]: This is a semver breaking change, we'll need to bump the major for the next release

Comment (DinosaurDad) [inline ARAnalytics.podspec]: It turned into quite a chain of required changes. I also noted that `xctool` no longer does builds, so I needed to do this:

```
xcodebuild -workspace Example/ARAnalyticsTests.xcworkspace -scheme ARAnalyticsBootstrapiOS -sdk iphonesimulator
xctool -workspace Example/ARAnalyticsTests.xcworkspace -scheme ARAnalyticsBootstrapiOS -sdk iphonesimulator run-tests
```

Comment (orta) [inline ARAnalytics.podspec]: If you give me access to this branch, I'll get it green 👍

Comment (DinosaurDad) [inline ARAnalytics.podspec]: You've got it. I'm also @Matt-ExpeData - the name that shows up depends on which app/machine I use for commits.

**Comment by orta** [comment]
Err - oops, do you still have those commits?

**Comment by orta** [comment]
I have what I think is a working `travis.yml` now:

```
language: objective-c
osx_image: xcode9
env:
  global:
    - LC_CTYPE=en_US.UTF-8
    - LANG=en_US.UTF-8
before_script:
  - echo -e "Host *\n\tStrictHostKeyChecking no\n" >> ~/.ssh/config
  - bundle exec danger
  - cd Example; pod install; cd ..

script:
  - set -o pipefail && xcodebuild -workspace Example/ARAnalyticsTests.xcworkspace -scheme ARAnalyticsBootstrapiOS -configuration Debug test -sdk iphonesimulator -destination platform='iOS Simulator',OS='9.0',name='iPhone 6' CODE_SIGNING_REQUIRED=NO | bundle exec xcpretty

```

**Comment by DinosaurDad** [comment]
I just force pushed. ;)

**Comment by orta** [comment]
I recreated this PR from scratch in #302 - will release that as 5.0.0

**Pull request closed without merging**